graylog2-web-interface/src/views/components/contexts/SearchPageLayoutProvider.tsx

```diff
@@ -14,46 +14,34 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
+
 import * as React from 'react';
-import { useMemo } from 'react';
 
-import SearchPageLayoutContext from './SearchPageLayoutContext';
-import SearchPageLayoutState from './SearchPageLayoutState';
+import type { LayoutState } from 'views/components/contexts/SearchPageLayoutContext';
+import SearchPageLayoutContext, { FULL_MENU } from 'views/components/contexts/SearchPageLayoutContext';
 
-type Props = {
+type SearchPageLayoutProviderProps = {
   children: React.ReactNode
-  setLayoutState: (stateKey: string, value: boolean) => void,
-  getLayoutState: (stateKey: string, defaultValue: boolean) => boolean,
-};
+  readonly providerOverrides?: LayoutState
+}
 
-const SearchPageLayoutProvider = ({ getLayoutState, setLayoutState, children }: Props) => {
-  const searchPageLayoutContextValue = useMemo(() => {
-    const config = {
-      sidebar: { isPinned: getLayoutState('sidebarIsPinned', false) },
-    };
-    const actions = { toggleSidebarPinning: () => setLayoutState('sidebarIsPinned', !config.sidebar.isPinned) };
+export const defaultState = {
+  sidebar: { isShown: true },
+  viewActions: FULL_MENU,
+} as LayoutState;
 
-    return ({
-      config,
-      actions,
-    });
-  }, [getLayoutState, setLayoutState]);
+function SearchPageLayoutProvider({ children, providerOverrides = defaultState }: SearchPageLayoutProviderProps) {
```

Comment:
Do we actually need this provider? If we set an appropriate default value for the context it becomes unnecessary.